### ISSUE TITLE
Add hydrostatic_stiffness matrix, to results dataset

### ISSUE THREAD

**Issue by davidfdomingos** [capytaine/capytaine#249]
Good morning, I would like to leave a suggestion to add the hydrostatic_stiffness matrix to the results of  "cpt.BEMSolver().fill_dataset(test_matrix_radiation, body)".
I have noticed when trying to compute RAOs that the hydrostatic_stiffness matrix needs to be manually added to these results. Which according to the tutorials is not the case when solving radiation and diffraction problems separately.

**Comment by mancellin** [comment]
Thank you for your comment.

There is something like that, if the body has an `hydrostatic_stiffness` parameter, it is added to the dataset:
```python
import xarray as xr
import capytaine as cpt

body = cpt.Sphere()
body.add_all_rigid_body_dofs()
body.center_of_mass = (0, 0, 0)
body.hydrostatic_stiffness = body.compute_hydrostatic_stiffness()
body.keep_immersed_part()

test_matrix = xr.Dataset(coords={"radiating_dof": ["Heave"], "omega": [1.0]})
ds = cpt.BEMSolver().fill_dataset(test_matrix, body)
print(ds["hydrostatic_stiffness"])
```
Is this what you have in mind?

Related maybe to #214, in which I'm trying to improve the way floating bodies are defined. In particular, I'm wondering if the hydrostatic stiffness should be automatically computed at the initialization of the body.

**Comment by davidfdomingos** [comment]
Thanks! Indeed "body.hydrostatic_stiffness = body.compute_hydrostatic_stiffness()" works, I was trying before with "body.compute_hydrostatics()", but this didn't work.

**Comment by mancellin** [comment]
Indeed, it would make sense for `body.compute_hydrostatic_stiffness()` to update the body to keep the matrices.
